clippy_lints/src/redundant_else.rs

```diff
@@ -138,3 +156,23 @@ impl BreakVisitor {
         self.check(stmt, Self::visit_stmt)
     }
 }
+
+// Extract the inner contents of an `else` block str
+// e.g. `{ foo(); bar(); }` -> `foo(); bar();`
+fn extract_else_block(mut block: &str) -> String {
+    block = block.strip_prefix("{").unwrap_or(block);
+    block = block.strip_suffix("}").unwrap_or(block);
+    block.trim_end().to_string()
+}
```

Comment:
We generally don't want to have out-parameters (even more so with output types) :cowboy_hat_face: 

```suggestion
fn extract_else_block(block: &str) -> String {
    let block = block.strip_prefix("{").unwrap_or(block);
    let block = block.strip_suffix("}").unwrap_or(block);
    block.trim_end().to_string()
}
```

Reply:
> We generally don't want to have out-parameters

Note that this is not an out parameter (this is `mut block: &str`, not `block: &mut str`).

Reply:
Just checked this, and it seems that indeed this isn't an out parameter!
We have no nits left. Everything's alright